app/import_functions_test.go

```diff
@@ -3386,6 +3491,70 @@ func TestImportImportDirectPost(t *testing.T) {
 		checkPreference(t, th.App, th.BasicUser2.Id, model.PreferenceCategoryFlaggedPost, post.Id, "true")
 	})
 
+	t.Run("Test with Type", func(t *testing.T) {
+		data := LineImportWorkerData{
+			LineImportData{
+				DirectPost: &DirectPostImportData{
+					ChannelMembers: &[]string{
+						th.BasicUser.Username,
+						th.BasicUser2.Username,
+					},
+					User:     ptrStr(th.BasicUser.Username),
+					Type:     ptrStr(model.PostTypeSystemGeneric),
+					Message:  ptrStr("Message with Type"),
+					CreateAt: ptrInt64(posttypeDate),
+				},
+			},
+			1,
+		}
+		errLine, err := th.App.importMultipleDirectPostLines(th.Context, []LineImportWorkerData{data}, false)
+		require.Nil(t, err)
+		require.Equal(t, 0, errLine)
+		AssertAllPostsCount(t, th.App, initialPostCount, 6, "")
+
+		posts, nErr := th.App.Srv().Store.Post().GetPostsCreatedAt(directChannel.Id, *data.DirectPost.CreateAt)
+		require.NoError(t, nErr)
+		require.Len(t, posts, 1)
+
+		post := posts[0]
+		require.Equal(t, post.Type, *data.DirectPost.Type)
+		require.Equal(t, post.Message, *data.DirectPost.Message)
+		require.Equal(t, post.CreateAt, *data.DirectPost.CreateAt)
+		require.Equal(t, post.UserId, th.BasicUser.Id)
```

Comment:
Wondering if these could be assertions? (I mean using `assert` package and letting the program continue in case of an error) This applies to other occurrences as well.

Reply:
Thanks for your comment. It makes sense, but let me ask you a bit more. The reason why I used `require` here is that I followed some existing test cases e.g.: ["Import the post again"](https://github.com/mattermost/mattermost-server/pull/18992/files#diff-d01a631362758f15aef5b2d85e9c97a3f625bc59468d45be50f60dc8f9dc84d9R3713). If I fix only the tests that I added, it looks inconsistent with the other tests around and looks weird. But if I fix all tests, it would be big changes.
Should I fix tests the I added or all tests?

Reply:
Good point @kaakaa, I think it would be fine for the tests cases you have added.

Reply:
Thanks. I've fixed this. If I misunderstand your thoughts, please letting me know it.
And I added [one more changes](https://github.com/mattermost/mattermost-server/pull/18992/commits/1b7c1a91ddcaef151822705ef1c934e1782a1fb6) for fixing CI failure.